### DISCUSSION TITLE
Can add an option which used to disable the host's hot key?

### DISCUSSION THREAD

**Discussion by HIDE-r** [rustdesk/rustdesk#4237] [General]
### Description

In my scenario. I use hyprland (A wayland compositor) for my window manager, and it have lots of hot key.

Is it possible to add an option for forwarding all keyboard input to the remote client? Now it always triggers my host's shortcut keys, which key is the same as my remote machine.

### Impact

who use linux window manager(dwm, awesomewm, hyprland...) like me.

**Comment by rustdesk** [comment]
Frankly impossible, wayland does not expose api to grab/trap global keyboard input. When you choose to use wayland, you have to face these limitation.

**Reply by oh-wind** [reply]
Wayland seems to have an [unstable protocol][1] that can achieve this.
Have you considered adding support for this protocol? It appears that most mainstream compositors support it.

[1]: https://wayland.app/protocols/keyboard-shortcuts-inhibit-unstable-v1

**Comment by fufesou** [comment]
https://github.com/rustdesk/rustdesk/pull/14302
